day1.js

```diff
@@ -0,0 +1,65 @@
+const formContainer = document.getElementById('formContainer');
+const bookTitle = document.getElementById('title');
+const bookAuthor = document.getElementById('author');
+const dynamicBooksDiv = document.getElementById('books-wrapper');
+
+const books = JSON.parse(localStorage.getItem('booksDetails')) || [];
+
+const addBook = (title, author, id) => {
+  books.push({
```

Comment:
- [ ] Your **`addBook`** method also has similar issues as **`deleteBook`**. As I add more books it begins adding books to the localStorage which I did not add myself. I believe this is because of the **books** array not accurately tracking the books. In order to work with a more accurate list of books, I would suggest you load your books from the localStorage directly in this array, push your new book to the loaded array and then update the localStorage with the new list like so:

  ```
  const loadedBooks = JSON.parse(localStorage.getItem('booksDetails')) || [];
  loadedBooks.push({
    title,
    author,
    id,
  });
  localStorage.setItem('booksDetails', JSON.stringify(loadedBooks));
  return { title, author, id };
  ```

Reply:
Hello @TedLivist thank you for this review.
I have looked at your suggestion and also looked at my previous code, there seems to be no difference apart from Vars. See it below:
```
const books = JSON.parse(localStorage.getItem('booksDetails')) || [];

const addBook = (title, author, id) => {
  books.push({
    title,
    author,
    id,
  });
  localStorage.setItem('booksDetails', JSON.stringify(books));
  return { title, author, id };
};
```
I have tried using your solution however and it still adds a new entry when the entry you are adding is very similar to the one that was there when you deleted it. which is not supposed to be the case. Kindly what could you suggest?

Reply:
@TedLivist I believe it was me who was trying it wrongly. I have rechecked and retried this way and it is working as expected. Thank you very much.
```
const addBook = (title, author, id) => {
  const loadedBooks = JSON.parse(localStorage.getItem('booksDetails')) || [];
loadedBooks.push({
  title,
  author,
  id,
});
localStorage.setItem('booksDetails', JSON.stringify(loadedBooks));
return { title, author, id };
};
```